test/unit_test/api/apps/restful_apis/test_retrieval_metadata_condition_pushdown.py

```
@@ -0,0 +1,183 @@
+import types
+from pathlib import Path
+
+import pytest
+
+sys.path.insert(0, str(Path(__file__).resolve().parents[5]))
+
+from common.metadata_utils import apply_meta_data_filter, convert_conditions
+
+
+class _FakePushdown:
+    calls: list[tuple[list[str], list[dict], str]] = []
```

Ruff (0.16.1)

[warning] Mutable default value for class attribute

(RUF012)